plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/ast/Exp.kt

```diff
@@ -407,6 +407,35 @@ sealed interface Exp : IntoSilver<viper.silver.ast.Exp> {
         override fun toSilver(): viper.silver.ast.Old = Old(exp.toSilver(), pos.toSilver(), info.toSilver(), trafos.toSilver())
     }
 
+    data class PredicateAccess(
+        val predicateName: MangledName,
+        val formalArgs: List<Exp>,
+        val pos: Position = Position.NoPosition,
+        val info: Info = Info.NoInfo,
+        val trafos: Trafos = Trafos.NoTrafos,
+    ) : Exp {
+        override val type: Type = Type.Bool
```

Comment:
Why is the type Bool?

Reply:
In Silver the type of `PredicateAccess` is `Bool`
https://github.com/viperproject/silver/blob/master/src/main/scala/viper/silver/ast/Expression.scala#L457

Reply:
Ah, okay, a comment might be good.  I suspect we will never use this type here, though, but I think unless we split things up we won't be able to capture that.